kmod/src/util.h

```diff
@@ -28,4 +28,50 @@ static inline long shrinker_min_long(long count)
 	return min(count, LONG_MAX);
 }
 
+
+/*
+ * We've tried to make a generic (off, len) overflow type detection
```

Comment:
gcc has an infinite precision __builtin_add_overflow() that could be used, and modern kernels have check_add_overflow().  We should be able to call check_add_overflow() without needing our own type wrappers.

Reply:
:clinking_glasses:

Reply:
Wasn't entirely straigthforward but a major improvement to begin. We have to create a tmp variable for the "d" parameter in the macro (it requires a "result" target variable and checks its type). And we have to still `-1` the len (carefully).